lib/matplotlib/axes/_axes.py

```diff
@@ -418,6 +418,21 @@ def legend(self, *args, **kwargs):
     def _remove_legend(self, legend):
         self.legend_ = None
 
+    @docstring.dedent_interpd
+    def colorbar(self, mappable, cax=None, use_gridspec=True, **kw):
+        """
+        Create a colorbar for a ScalarMappable instance, *mappable*, with the
+        axes as the parent.
+
+        Documentation for the pyplot thin wrapper:
+        %(colorbar_doc)s
+        """
+        ax = kw.pop('ax', None)
+        if ax is not None:
+            warnings.warn('Supplied ax argument ignored')
```

Comment:
I would just make this an error, not much point of being tolerant in a new API...

Reply:
I'd say it pretty much depends on how the documentation is written. If it says "allowed kwargs are all `matplotlib.figure.Figure.colorbar` arguments", then it should warn. If it explicitely says "allowed kwargs are all `matplotlib.figure.Figure.colorbar` arguments, *except*  `ax`", it can error out.

Reply:
:+1: to raising.  In someplaces we have to tolerate ignoring input because it is old API, but we should not add ignored kwargs in new API.